From: Ben Hutchings <ben@decadent.org.uk>
Date: Thu, 29 Sep 2016 18:00:41 +0100
Subject: [PATCH 395/397] genirq: Fix pointer cast warning for randomness on
 32-bit architectures
Origin: https://git.kernel.org/cgit/linux/kernel/git/rt/linux-stable-rt.git/commit?id=d53800e3bde2b76e0c3de4df5bc991bfb3a51d28

gcc warns about casting directly from a 32-bit pointer to u64.  In the
current rt patch series we cast to unsigned long to avoid the warning,
but this fix hasn't been backported to all versions.

Link: http://lkml.kernel.org/r/20160929170041.GW10601@decadent.org.uk

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
Signed-off-by: Steven Rostedt <rostedt@goodmis.org>
---
 kernel/irq/manage.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/kernel/irq/manage.c b/kernel/irq/manage.c
index c4677795d7c8..b1683ba80c88 100644
--- a/kernel/irq/manage.c
+++ b/kernel/irq/manage.c
@@ -897,7 +897,8 @@ static int irq_thread(void *data)
 #ifdef CONFIG_PREEMPT_RT_FULL
 			migrate_disable();
 			add_interrupt_randomness(action->irq, 0,
-						 desc->random_ip ^ (u64) action);
+						 desc->random_ip ^
+						 (unsigned long) action);
 			migrate_enable();
 #endif
 		}
